### PR TITLE
Changed sysctl configuration to exclusively use templates.

### PR DESCRIPTION
**Pull request by KoenDG**
Ok, I said earlier I'd take a look next week, but I found I had nothing to do this evening, so I had a look and came up with this.

This code provides two templates: one for generic sysctl hardening, and an other for disabling ipv6.

In my opinion, this separates the concerns of generic sysctl hardening versus disabling ipv6. Both can be done on their own, and this way both can have their appropriate tags, and only those, without extras that only apply in certain situations.

This also works around an other issue where systemd's sysctl binary doesn't have the option to load only a single file with sysctl settings. You're stuck restarting the service completely, even if you only want to change 1 file. By splitting this into 2 config files, one doesn't hinder the other.

On top of that, I've added code to remove the ipv6 sysctl file, if the system doesn't have ipv6 support. This can happen if a previous run of these playbooks(or whatever the user did to their device manually) disabled it, and now there's a file there that doesn't apply.

I'm not sure it's strictly necessary to remove it, though it seems logically consistent. Maybe it's more pragmatic to leave it in place?

Which brings me to 2 final question:

1. Do we want to remove the ipv6 conf file in case the machine doesn't have ipv6 anymore? I'm not sure if it's necessary to leave it in place, after the GRUB settings have been altered and the system has been rebooted.
2. Are there any tags missing for either case?

I don't think "what if the user wants to re-enable ipv6" is a relevant question for this project, as if focuses on hardening only, not undoing the hardening.

So maybe 3rd question: should something to that effect be added to the readme? Or maybe it's already there, it's been a while since I checked.

I'm not sure about what the linter is going to say about what I've done, so that and any other remarks are welcome.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -812,26 +812,31 @@ common_account_template: etc/pam.d/common-account.j2
...
 ssh_config_template: etc/ssh/ssh_config.j2
 sshd_config_template: etc/ssh/sshd_config.j2
+sysctl_ipv6_config_template: etc/sysctl/sysctl.ipv6.conf.j2
+sysctl_main_config_template: etc/sysctl/sysctl.main.conf.j2
 system_conf_template: etc/systemd/system.conf.j2
 timesyncd_conf_template: etc/systemd/timesyncd.conf.j2
-useradd_template: etc/default/useradd.j2
+tmp_mount_template: etc/systemd/tmp.mount.j2
 user_conf_template: etc/systemd/user.conf.j2
+useradd_template: etc/default/useradd.j2
```
Comment (KoenDG) [inline README.md:839]: I copied the list straight from the templates.yml file.

**Comment by KoenDG** [comment]
Cleanup, typos, squashed into 1 commit, should be done now, from my point at this moment.

File: README.md (hunk 1 above)
Comment (konstruktoid) [reply README.md:839]: yeah, the README is (almost) autogenerated

File: tasks/ipv6.yml [hunk 2]
```diff
@@ -27,19 +27,24 @@
...
+    owner: root
+    group: root
+    backup: false
+  when: system_has_ipv6
   notify:
     - Restart sysctl
   tags:
     - ipv6
+
+- name: Make sure we delete the ipv6 sysctl file if this machine doesn't support ipv6
```
Comment (konstruktoid) [inline tasks/ipv6.yml:45]: This should trigger the `Restart sysctl` handler

File: templates/etc/sysctl/sysctl.ipv6.conf.j2 [hunk 3]
```diff
@@ -0,0 +1,8 @@
+### IPV6 related settings
```
Comment (konstruktoid) [inline templates/etc/sysctl/sysctl.ipv6.conf.j2:1]: Add comments
```
# {{ ansible_managed }}
# Generated by Ansible role {{ ansible_role_name }}
```

File: templates/etc/sysctl/sysctl.main.conf.j2 [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{% if (ansible_kernel is version("5", ">=")) %}
```
Comment (konstruktoid) [inline templates/etc/sysctl/sysctl.main.conf.j2:1]: Add comments
```
# {{ ansible_managed }}
# Generated by Ansible role {{ ansible_role_name }}
```

**Review by konstruktoid — CHANGES_REQUESTED** [review]

**Comment by konstruktoid** [comment]
> 1. Do we want to remove the ipv6 conf file in case the machine doesn't have ipv6 anymore? I'm not sure if it's necessary to leave it in place, after the GRUB settings have been altered and the system has been rebooted.

I'm fine with removing it, since it will be added later if requried.

> 2. Are there any tags missing for either case?

No worries, I'm planning to add and remove some later onanyway

> So maybe 3rd question: should something to that effect be added to the readme? Or maybe it's already there, it's been a while since I checked.

I'll update the README after this has been merged.

> I'm not sure about what the linter is going to say about what I've done, so that and any other remarks are welcome.

The linter was happy :+1:

File: templates/etc/sysctl/sysctl.main.conf.j2 (hunk 4 above)
Comment (KoenDG) [reply templates/etc/sysctl/sysctl.main.conf.j2:4]: Done

File: templates/etc/sysctl/sysctl.ipv6.conf.j2 (hunk 3 above)
Comment (KoenDG) [reply templates/etc/sysctl/sysctl.ipv6.conf.j2:4]: Done

**Comment by KoenDG** [comment]
Updated as requested.

**Comment by konstruktoid** [comment]
Thanks @KoenDG!

**Pull request merged**